src/main/java/com/spm/vasylyshyn/service/UserService.java

```diff
@@ -102,55 +140,41 @@ private User getUserByPrincipal(Principal principal) {
                 .orElseThrow(() -> new UsernameNotFoundException("User not found with username " + username));
```

Comment:
Maybe the constructor of your exceptions should take actual variable data as parameters, and add "User not found with username" boilerplate code inside the implementation?

Reply:
Implement it